lib/runner/extensions/request.command.js

```diff
@@ -138,36 +57,32 @@ module.exports = {
          * @todo  validate payload
          */
         request: function (payload, next) {
-            // @todo - resolve variables in a more graceful way
-            // @todo - no need to sync vatiables when SDK starts supporting resolution from scope directly
-            var item = new sdk.Item(payload.item.toObjectResolved(null,
-                [payload.data, payload.environment.syncVariablesTo({}), payload.globals.syncVariablesTo({})])),
-
-                abortOnError = _.has(payload, 'abortOnError') ? payload.abortOnError : this.options.abortOnError,
+            var abortOnError = _.has(payload, 'abortOnError') ? payload.abortOnError : this.options.abortOnError,
                 self = this;
 
             async.waterfall([
-                // Process any authentication helpers in the request.
+                // Run the helper functions
+                // todo - move this to the right place
                 function (cb) {
-                    // Re-parse the URL, because variables have been resolved now, and things might be moved around
-                    item.request.url = new (sdk.Url)(item.request.url.toString());
-                    cb(null, item.request.authorize());
-                },
-                // Handle file resolution
-                function (request, cb) {
-                    var resolver = _.get(self.options, 'fileResolver') ||
-                        // todo: remove this once apps give the options correctly
-                        _.get(self.options, 'requester.fileResolver');
-                    resolveFiles(request, resolver, self.triggers, payload.coords, cb);
+                    async.series(
```

Comment:
there should be some easier async function - I guess async.apply ...
Also, no need to rebind here. These need to be super performant.